README.md

```diff
@@ -567,7 +635,160 @@ export default class ScrollPositionModifier extends Modifier<ScrollPositionModif
 }
 ```
 
-See [this pull request comment](https://github.com/sukima/ember-class-based-modifier/pull/5#discussion_r326687943) for a full discussion about using TypeScript with your Modifiers.
+However, while doing so is slightly more convenient, it means you get *much worse* feedback in tests or at runtime if someone passes the wrong kind of arguments to your modifier.
+
+### Lifecycle hooks and types
+
+It is impossible to correctly type the `element` in the current design of the class-based modifier: as noted in [the lifecycle hooks discussion](#lifecycle-summary), `element` is unavailable during `constructor` and `willDestroy`. Currently, `element` is typed as `Element`, since that is correct for *most* places in the lifecycle of the app. However, this means that you must take care to remember that the type is *wrong* in `constructor` and `willDestroy`.
+
+We cannot fix this *at all* for `constructor` with the current definition. However, to work around this in `willDestroy`, we supply a utility type, `InTeardown`, which you can use to specify the type of `this`. For example:
+
+```ts
+import Modifier, { InTeardown } from 'ember-modifier';
+
+function doSomethingWithElement(el: Element) {
+  // ...
+}
+
+export default class CorrectlyTypedModifier extends Modifier {
+  willRemove(this: InTeardown<CorrectlyTypedModifier>) {
```

Comment:
what are the odds the TS team lets this be inherited from the super class?

Reply:
Heh, almost certainly not—it'd be pretty weird, and likely broken, in part because you'd end up throwing away data about the subclass… and since a `Parent` almost always has fewer properties, methods, etc. than does `Subclass`, that'd be a pretty terrible outcome. (I actually experimented with that the other day, realized it didn't work and then had to think through *why* it didn't.)